### DISCUSSION TITLE
RegExp in block

### DISCUSSION THREAD

**Discussion by prebenzacho** [uNmAnNeR/imaskjs#945] [Q&A]
Can someone elaborate why this one works:

```
var mask = IMask(document.getElementById("testid"), {
  mask: '$num',
  blocks: {
    num: {
      mask: new RegExp("^[0-9]+$"),
    },
  }
});
```

While this one does not:

```
var mask = IMask(document.getElementById("testid"), {
  mask: '$num',
  blocks: {
    num: {
      mask: new RegExp("^[0-9]{3}$"),
    },
  }
});
```

I have tested the RegEx in a regex editor, should be a valid match. But I can not enter anything in the input field in the last example, but in the first example I can enter digits only, as expected.

**Comment by uNmAnNeR** [comment]
from docs:
![image](https://github.com/uNmAnNeR/imaskjs/assets/1927921/dabac077-7922-4262-a1e2-7ecfdabae4aa)
